drivers/spi/spi_nrfx_common.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "spi_nrfx_common.h"
+#include <zephyr/kernel.h>
+#include <nrfx_gpiote.h>
+
+int spi_nrfx_wake_init(uint32_t wake_pin)
+{
+	nrfx_gpiote_input_config_t input_config = {
+		.pull = NRF_GPIO_PIN_PULLDOWN,
+	};
+	uint8_t ch;
+	nrfx_gpiote_trigger_config_t trigger_config = {
+		.trigger = NRFX_GPIOTE_TRIGGER_HITOLO,
+		.p_in_channel = &ch,
+	};
+	nrfx_err_t res;
+
+	res = nrfx_gpiote_channel_alloc(&ch);
+	if (res != NRFX_SUCCESS) {
+		return -ENODEV;
+	}
+
+	res = nrfx_gpiote_input_configure(wake_pin,
+					  &input_config,
+					  &trigger_config,
+					  NULL);
+	if (res != NRFX_SUCCESS) {
+		nrfx_gpiote_channel_free(ch);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int spi_nrfx_wake_request(uint32_t wake_pin)
+{
+	nrf_gpiote_event_t trigger_event = nrfx_gpiote_in_event_get(wake_pin);
+	uint32_t start_cycles;
+	uint32_t max_wait_cycles =
+		DIV_ROUND_UP(CONFIG_SPI_NRFX_WAKE_TIMEOUT_US *
+				 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
+			     1000000);
```

Comment:
This hardcoded 1000000 will make the time different depending on target clock frequency? Or am I missing something?

Reply:
Ah, now I get it. The 1000000 is just number of us in a second.

Reply:
This is <timeout_us> * <hw_cycles_per_sec>/1000000.
The timeout value is in microseconds so it needs to be multiplied by the number of cycles per second divided by 1000000 to get the number of cycles to wait.